src/dune_rules/merlin/merlin.ml

```diff
@@ -390,8 +390,10 @@ module Unprocessed = struct
       | Some args ->
         let action =
           let action = Preprocessing.chdir (Run (exe, args)) in
-          Action_unexpanded.expand_no_targets ~loc ~expander ~deps:[]
+          Action_unexpanded.expand_no_targets ~dir ~loc ~expander ~deps:[]
             ~what:"preprocessing actions" action
+            ~lib_name (* TODO: Need ml_kind for merlin support  *)
```

Comment:
could you try adding merlin support?

Reply:
Or actually, does it? Doesn't merlin decide to pass `--intf` or `--impl`?

@jchavarri did you check if this PR works with merlin?

Reply:
It won't work with merlin yet because we're not generating the `ppx` directive. e.g. `progn` falls here: https://github.com/ocaml/dune/blob/a5fd306a4e24c7bd67330b411b3e5662b7e03039/src/dune_rules/merlin/merlin.ml#L409

> I think this still needs to be addressed, right?

I thought we had decided to add it in a follow-up PR, hence the `TODO`.